.github/workflows/unit_test_pipeline.yml

```diff
@@ -0,0 +1,66 @@
+name: TM1Py Integration Tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      environments:
+        description: 'JSON array of environments to test (e.g., ["tm1-11-onprem", "tm1-11-cloud"])'
+        required: true
+        default: '["tm1-11-onprem","tm1-11-cloud","tm1-12-mcsp"]'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        environment: ${{ fromJson(github.event.inputs.environments) }}
+    environment: ${{ matrix.environment }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install dependencies
+        run: |
+          pip install -e .[unit_test]
+
+      - name: Retrieve TM1 Connection Details
+        id: get-connection
+        run: echo "Retrieving TM1 connection details"
+        env:
+          TM1_CONNECTION: ${{ vars.TM1_CONNECTION }}
+          TM1_CONNECTION_SECRET: ${{ secrets.TM1_CONNECTION_SECRET }}
+
+      - name: Generate config.ini
+        run: |
+          python Tests/resources/generate_config.py
+        env:
+          TM1_CONNECTION: ${{ vars.TM1_CONNECTION }}
+          TM1_CONNECTION_SECRET: ${{ secrets.TM1_CONNECTION_SECRET }}
```

Comment:
Yes, it would be the same result, but the process would be reversed. I think it would be more common to use the environment variables directly and read them from a file for development purposes.

Reply:
I'm not sure I'm following your proposal, could you elaborate a bit more?

Reply:
For sure. At the moment we have got two cases:
1) A developer creates a test/config.ini 
2) The CI Pipeline creates a test/config.ini based on the GitHub Variables 

My proposal is to change how the test get their configuration.  So instead of reading the config.ini, I would read environment variables in the tests. 

To simplify local development (case 1), the parameters from config.ini should be read and set as envs. 

I hope it's clearer now. These are just a few small changes. I could make PR draft for further discussion.

Reply:
@onefloid I think long term we can substitute the `.ini` with `.env` and read environment variables in the tests instead of the `ini`.

I think for the moment Nico's solution solved `.ini` file creation quite elegantly.

Reply:
Yes, I agree. This is not super urgent and good for another PR.